### PR TITLE
Add Luminances and Scheme pages

### PR DESCRIPTION
**Pull request by attitude**
Add internal documentation & helper components to generate luminance scale & test schemes.

---
This change is [Reviewable](https://reviewable.io/reviews/contember/docs/52)

### CODE REVIEW

File: src/theming/Luminance/Chart.tsx [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+import { interpolateBasis, quantize } from "d3-interpolate"
+import * as React from "react"
+import { Line } from "react-chartjs-2"
+
+ChartJS.register(CategoryScale, LineElement, PointElement, LinearScale, Title);
+
+const COLORS_COUNT = 41;
+const COLORS_RANGE = Array.apply(null, Array(COLORS_COUNT))
+  .map((_, index) => index)
+  .reverse();
```
Comment (jonasNobile) [inline src/theming/Luminance/Chart.tsx:15]:
```suggestion
const COLORS_RANGE = Array.from({length: COLORS_COUNT}, (_, index) => index).reverse()
```

File: src/theming/Luminance/Chart.tsx [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+const COLORS_X_AXIS = COLORS_RANGE.map(
+  (index) => (1000 - (1000 / (COLORS_COUNT - 1)) * index) / 1000
+);
+
+const count = 41;
+const range = Array.apply(null, Array(count))
+  .map((_, index) => index)
+  .reverse();
```
Comment (jonasNobile) [inline src/theming/Luminance/Chart.tsx:24]:
```suggestion
const range = Array.from({length: count}, (_, index) => index).reverse()
```

File: src/theming/Luminance/Chart.tsx [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+      <pre
+        style={{
+          color: "white",
+          background: "black",
+          padding: "0.5em",
+          position: "sticky",
+          top: 0
+        }}
+      >
+        {"\n"}
```
Comment (jonasNobile) [inline src/theming/Luminance/Chart.tsx:101]:
```suggestion
        <br/>
```

File: src/theming/Luminance/Chart.tsx [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+        Luminance values:{" "}
+        {JSON.stringify(
+          luminancesList
+            .map((number) => Math.round(number * 10000) / 10000)
+            .filter((number, index) => index % 5 === 0)
+        )}
+        {"\n"}
```
Comment (jonasNobile) [inline src/theming/Luminance/Chart.tsx:108]:
```suggestion
        <br />
```

**Review by jonasNobile — CHANGES_REQUESTED** [review]

File: src/theming/Helpers/rangeOfNumbers.ts [hunk 5]
```diff
@@ -0,0 +1,6 @@
+export function rangeOfNumbers (start: number, end: number, step: number = 1) {
```
Comment (jonasNobile) [inline src/theming/Helpers/rangeOfNumbers.ts:1]:
```suggestion
export function rangeOfNumbers(start: number, end: number, step: number = 1) {
```

**Comment by VojtaStanek** [comment]
The `rangeOfNumbers` func is:
a) too complicated
b) wrong (eg. why round?)

**Comment by attitude** [comment]
@VojtaStanek, `rangeOfNumbers()` does suit its needs. I refactored the code to be more readable and fixed use of `Math.round()` to `Math.floor()` method and confirmed it is working. Can you be now more specific about your concerns?

(Please note, there is only a slight change this method would be use otherwise as internal for the docs.)

Thanks

**Comment by VojtaStanek** [comment]
Looks ok now.

**Review by jonasNobile — CHANGES_REQUESTED** [review]
> Reviewed 28 of 28 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/contember/docs/52)* status: all files reviewed, 1 unresolved discussion (waiting on @attitude)

___
*[src/theming/Helpers/cssToSASS.ts, line 2 at r1](https://reviewable.io/reviews/contember/docs/52#-N0Fqsj3D6gidb9LYyqA:-N0Fqsj4-XVcyOBariwF:bz916y2) ([raw file](https://github.com/contember/docs/blob/b41d84d8cfbc9c0f28071672ddcc2f42f6961d3d/src/theming/Helpers/cssToSASS.ts#L2)):*
> ```XML
> export function cssToSASS(css: string) {
>   return css.replace(/;/g, "").replace(/\{/g, "").replace(/\}/g, "").trim()
> ```

OR

```ts
css.replaceAll(";", "").replaceAll("{", "").replaceAll("}", "").trim()
```



_Suggestion:_
```XML
  return css.replace(/[;{}]/g, "").trim()
```

**Review by attitude — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/contember/docs/52)* status: 5 of 28 files reviewed, 1 unresolved discussion (waiting on @jonasNobile)

___
*[src/theming/Helpers/cssToSASS.ts, line 2 at r1](https://reviewable.io/reviews/contember/docs/52#-N0Fqsj3D6gidb9LYyqA:-N0FxuYh8WkafLsW17s2:b-896fix) ([raw file](https://github.com/contember/docs/blob/b41d84d8cfbc9c0f28071672ddcc2f42f6961d3d/src/theming/Helpers/cssToSASS.ts#L2)):*
<details><summary><i>Previously, jonasNobile (Jonáš Nobile) wrote…</i></summary><blockquote>

OR

```ts
css.replaceAll(";", "").replaceAll("{", "").replaceAll("}", "").trim()
```
</blockquote></details>

Done.

**Review by jonasNobile — APPROVED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/contember/docs/52)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @attitude)

**Pull request merged**